kt/godot-library/godot-coroutine-library/src/main/kotlin/godot/coroutines/awaitResourceLoad.kt

```diff
@@ -1,48 +1,32 @@
 package godot.coroutines
 
-import godot.core.Error
 import godot.api.Resource
 import godot.api.ResourceLoader
 import godot.api.ResourceLoader.CacheMode
-import godot.core.variantArrayOf
-import godot.global.GD
-import godot.common.util.RealT
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.Job
 import kotlinx.coroutines.async
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.runBlocking
-import kotlinx.coroutines.suspendCancellableCoroutine
-import kotlin.coroutines.resume
-import kotlin.coroutines.resumeWithException
 
 
 /**
  * Suspends the coroutine until a resource is loaded from the given path.
  *
  * @param path The path of the resource to be loaded.
  * @param typeHint A hint about the type of resource being loaded.
- * @param useSubThreads Specifies whether to use sub-threads for loading the resource.
  * @param cacheMode The cache mode to be used while loading the resource.
- * @param onProgress A callback function to track the progress of resource loading.
  * @return The loaded resource, or null if there was an error.
  */
 public suspend inline fun ResourceLoader.awaitLoad(
     path: String,
     typeHint: String = "",
-    useSubThreads: Boolean = false,
-    cacheMode: CacheMode = ResourceLoader.CacheMode.REUSE,
-    crossinline onProgress: (RealT) -> Unit = {},
```

Comment:
I'd love for this to make a return at some point :-) Just a note not really related to this PR

Reply:
The parameters were actually unused, we forgot to remove them when we realised we couldn't use the threaded requests for the ResourceLoader.
As a reminder, despite being threaded, it doesn't support multiple threads requesting the same resource at the same time >< which is an issue when we try to use coroutines to load several instances of the same scene.